functions/supplier/zeromq-supplier/src/main/java/org/springframework/cloud/fn/supplier/zeromq/ZeroMqSupplierProperties.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.supplier.zeromq;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import org.hibernate.validator.constraints.Range;
+import org.zeromq.SocketType;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.util.Assert;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ *
+ * @author Daniel Frey
+ * @since 3.1.0
+ */
+@ConfigurationProperties("zeromq.supplier")
+@Validated
+public class ZeroMqSupplierProperties {
+
+	private static final SocketType DEFAULT_SOCKET_TYPE = SocketType.SUB;
+	private static final Duration DEFAULT_CONSUME_DELAY = Duration.ofSeconds(1);
+	private static final List<SocketType> VALID_SOCKET_TYPES =
+			Arrays.asList(SocketType.PAIR, SocketType.PULL, SocketType.SUB);
+
+	private final AtomicInteger bindPort = new AtomicInteger();
+
+	/**
+	 * The {@link SocketType} the connection should make.
+	 *
+	 * Allowable values:
+	 * - PAIR
+	 * - PULL
+	 * - SUB
+	 */
+	private SocketType socketType = DEFAULT_SOCKET_TYPE;
+
+	/**
+	 * Connection URL for {@link org.zeromq.ZMQ.Socket#connect(String)}.
+	 * Mutually exclusive with the {@link #bindPort}.
+	 */
+	private String connectUrl;
+
+	/**
+	 * A {@link Duration} to delay consumption when no data received.
+	 */
+	private Duration consumeDelay = DEFAULT_CONSUME_DELAY;
+
+	/**
+	 * Topics the {@link SocketType#SUB} socket is going to use for subscription.
+	 * It is ignored for all other {@link SocketType}s supported.
+	 */
+	private String[] topics = {""};
+
+	/**
+	 * @param socketType the {@link SocketType} to establish,
+	 *                      defaults to {@link #DEFAULT_SOCKET_TYPE} if empty
+	 */
+	@NotNull(message = "'socketType' is required")
+	public void setSocketType(SocketType socketType) {
+		Assert.state(VALID_SOCKET_TYPES.contains(socketType),
```

Comment:
You don't need this logic here. It is just a matter of channel adapter functionality and it is validated over there.